elastic/kibana
# [Infra] Display friendlier Lens generated error messages

closes [#189064](https://github.com/elastic/kibana/issues/189064)

## Summary

This PR changes Lens embeddable to support creating custom error messages by using the new `customBadgeMessages` prop.

<img width="500px" alt="image" src="https://github.com/user-attachments/assets/d48d86fa-b4a7-4dd5-ad84-9fe4bf144672">

The custom error message will only be displayed in the context of the hosts view and asset details, which is where this new prop is being used. Everywhere else will display the default error handling provided by Lens

<img width="500px" alt="image" src="https://github.com/user-attachments/assets/38ecaee9-5f25-4d34-85e4-f176095982c5">


### How to test
- Start a local kibana and es instances
- Run `node scripts/synthtrace infra_hosts_with_apm_hosts --live`
- Change the the index patter in Settings to `metrics-apm`
- Navigate to Infrastructure > Hosts View and open the asset details flyout
  - The missing field message should be displayed as the screenshot above 
- Open a chart in lens
  - The default Lens message will be shown
